lib/facade_converter.ts

```diff
@@ -354,6 +396,48 @@ export class FacadeConverter extends base.TranspilerBase {
     },
   };
 
+  private es6Promises: ts.Map<CallHandler> = {
+    'Promise.resolve': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.value(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise.reject': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.error(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise': (c: ts.CallExpression, context: ts.Expression) => {
+      if (c.kind != ts.SyntaxKind.NewExpression) {
+        return true;
+      }
+      this.assert(c.arguments.length == 1);
+      const callback = <ts.ArrowFunction>(c.arguments[0]);
+      this.assert(
+          callback.kind == ts.SyntaxKind.ArrowFunction ||
+          callback.kind == ts.SyntaxKind.FunctionDeclaration ||
+          callback.kind == ts.SyntaxKind.FunctionExpression);
+      this.assert(callback.parameters.length > 0 && callback.parameters.length < 3);
+
+      this.emit('(() {');  // Create a new scope.
+      this.emit('Completer _completer = new Completer();');
```

Comment:
This code is too clever. You also risk generating code that's substantially larger than the original source.

I think we should use a helper method for this.